### ISSUE TITLE
Question about datatype of metrics

### ISSUE THREAD

**Issue by nginn** [dalmatinerdb/dalmatinerdb#125]
Hello,

Can I write NaN as one of the possible metric values to dalmatinerdb and get it as a result query?

Explanation of use case: I have some metric X and I want to distinguish between the situation when I have _collected metric but failed to write_ this metric to dalmatinerdb (so it will store NaN itself and then return me approximation of value for particular datapoint with lower confidence value) and when I _can't collect that metric but I'm perfectly able to write_ to dalmatinerdb.

If it's not possible I still can write some not ordinary values for metric (like -1 for unsigned int metric) but NaN option seems like a better way to go.

Thanks!

**Comment by Licenser** [comment]
Hi,
dalmatinerdb stores all data in what you can think about a huge distributed array per series. So if you write `s[0]` and `s[2]`, we keep track of `s[1]` never been written (what you call NaN). The query engine will try to intrapolate the value with the very simple assumption that if `s[0]` was set that `s[1]` will not have changed much. However as part of the query engine DlamtinerDB uses something we call 'confidence' that keeps track of how much of the data a value consusts of is extrapolated and how much really recored. so the confidence for `s[0]` and `s[2]` would be `1.0` while for `s[1]` it remains `0.0` even if it is displayed as the same value as `s[0]`, confidence does get propagated throughout aggregates (so the confidence of `s[0] + s[1]` would be `0.5` and so on).

So in short: Not writing == writing NaN
